### DISCUSSION TITLE
Provide dedicated SEO tag management inbuilt to this package

### DISCUSSION THREAD

**Discussion by modbender** [vercel/next.js#43519] [Ideas]
### Describe the feature you'd like to request

Management of SEO tags must be a out of the box implementation that should come with this package as it is an important part of any website and overall concept of Web.

### Describe the solution you'd like

Rather than writing each meta tag manually under the `Head` component provided by next (keeping this as an option), a separate tag similar to `NextSeo` tag provided by `next-seo` package can be added.

But this would probably still not be enough of a reason to introduce a separate tag, but what I'm aiming for here is something similar to what `Nuxtjs` is providing

An example I was imagining which could be provided by nextjs in server side or static data fetching

```js
export async function getServerSideProps(context) {
  const res = await fetch(`https://.../data`);
  const data = await res.json();

  if (!data) {
    return {
      notFound: true,
    };
  }

  return {
    props: { data }, // will be passed to the page component as props
    head: {
      title: data.title,
      siteName: 'SiteName',
      description: data.description,
      canonical: "https://www.mydomain.com" + context.resolvedUrl,
      images: [
        {
          url: "https://www.mydomain.com" + data.image.path,
          width: 800,
          height: 600,
          alt: "Og Image Alt",
          type: "image/jpeg",
        },
      ],
      twitter: {
        handle: '@handle',
        site: '@site',
        cardType: 'summary_large_image',
      },
    },
  };
}
```

### Describe alternatives you've considered

The alternative I've been using is [next-seo](https://www.npmjs.com/package/next-seo) package, but it can be better if this is internal.

**Comment by GabenGar** [comment]
SEO-related development is a complete mess also. Adding it to upstream will only increase entropy of NextJS development. And can also lead to quite real damage in the form of ranking penalties caused by mistakes in upstream.

**Reply by modbender** [reply]
making it dev opt able part like `next/seo` won't hurt.

**Comment by erwanvivien** [comment]
I believe having a simple api to generate all the meta tags would be nice!

Having to deal with this by hand is long and uninteresting. Such a feature would be amazing.
